sale_coupon_domain_product_reward_fixed_price/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _set_reward_fixed_price_for_lines(self, program):
+        """
+        Update discount field by program
+        """
+        price_unit = program.price_unit
+        lines = self._get_paid_order_lines()
+        if (
+            program.discount_apply_on == "specific_products"
+            and program.discount_apply_on_domain_product
+        ):
+            lines = (self.order_line - self._get_reward_lines()).filtered(
+                lambda line: program._get_valid_products(line.product_id),
+            )
+            for line in lines:
```

Comment:
Not need use cycle here. You must use `write` method.
`lines.write({"price_unit": price_unit})`

Reply:
Thanks, I'll fix it)